### PR TITLE
Removed Immutable and resolved flow issues

### PR DESCRIPTION
**Pull request by chandrakailash1993**
Did Following set of Tasks.

1) removed immutable from the application.
2) Added missing packages eg: 'invarient', 'isomorphic-unfetch','stylelint-processor-styled-components' etc...
3) Resolved few styled component issues
4) Resolved many flow related issues.

### CODE REVIEW

File: app/components/atoms/HeadTag/HeadTag.js [hunk 1]
```diff
@@ -1,33 +1,23 @@
...
+    content: { ...props }.content,
   };
-  // eslint-disable-next-line
-  attrs[props.prefix && props.prefix === 'og' ? 'property' : 'name'] = props.prefix
-    ? `${props.prefix}:${props.meta.key}`
-    : props.meta.key;
+
+  /* eslint-disable */
+  // $flow-disable-line
+  attrs[{ ...props }.prefix && { ...props }.prefix === 'og' ? 'property' : 'name'] = { ...props }
```
Comment (amrit07) [inline app/components/atoms/HeadTag/HeadTag.js]: what's the diff? props.prefix vs {...props}.prefix; They result in same but extra processing is required in the former; any benefits?

File: app/components/atoms/Iframe/Iframe.js [hunk 2]
```diff
@@ -57,12 +65,12 @@ class Iframe extends PureComponent<Props, State> {
...
         src={sourceUrl}
         width={width}
-        height={this.props.takeContentHeight ? this.state.frameHeight : height}
+        height={takeContentHeight ? frameHeight : height}
         frameBorder={frameBorder}
         onLoad={event => {
           if (takeContentHeight) {
             this.setState({
-              frameHeight: `${this.iframe.contentWindow.document.body.scrollHeight}px`,
+              frameHeight: `${contentWindow.document.body.scrollHeight}px`,
```
Comment (amrit07) [inline app/components/atoms/Iframe/Iframe.js]: code will break when contentWindow is undefined

File: app/components/molecules/Testimonials/Testimonials.js [hunk 3]
```diff
@@ -36,7 +36,9 @@ class Testimonials extends PureComponent<Props> {
                     <img src="/static/images/pic01.jpg" alt="" />
                   </div>
                   <p className="credit">
-                    - <strong>Jane Doe</strong> <span>CEO - ABC Inc.</span>
+                    <span>-</span>
```
Comment (amrit07) [inline app/components/molecules/Testimonials/Testimonials.js]: is empty span intended one?

File: app/components/templates/ErrorPage/ErrorPage.js [hunk 4]
```diff
@@ -31,10 +33,11 @@ class ErrorPage extends Component<any> {
 
 /* istanbul ignore next */
 const mapStateToProps = state => ({
-  errorData: state.getIn(['errorPage', 'errorPageData']),
+  state,
+  errorData: get(state, ['errorPage', 'errorPageData']),
 });
 
-const mapDispatchToProps = (dispatch: Map) => ({});
+const mapDispatchToProps = () => ({});
```
Comment (amrit07) [inline app/components/templates/ErrorPage/ErrorPage.js]: we can remove this unnecessary mapDispatchToProps

File: app/components/templates/HomePage/HomePage.js [hunk 5]
```diff
@@ -11,9 +11,9 @@ import Highlight from '../../molecules/Highlights';
 import Testimonials from '../../molecules/Testimonials';
 import TopBanner from '../../molecules/TopBanner';
 
-type Props = {};
+import type { PropsTypes } from './types';
```
Comment (amrit07) [inline app/components/templates/HomePage/HomePage.js]: Let's follow the consistent naming convention unless we require special name use Props for propsType

File: app/lib/ContentSlotUnsafeHTML/ContentSlotUnsafeHTML.js [hunk 6]
```diff
@@ -44,20 +44,25 @@ class ContentSlotUnsafeHTML extends PureComponent<Props> {
   };
 
   componentDidMount = () => {
-    if (this.props.hasScript && document && document.createRange) {
+    const { hasScript } = this.props || undefined;
```
Comment (amrit07) [inline app/lib/ContentSlotUnsafeHTML/ContentSlotUnsafeHTML.js]: no need for extra || undefined ; will default to that automatically

File: app/components/atoms/HeadTag/HeadTag.js (hunk 1 above)
Comment (chandrakailash1993) [reply app/components/atoms/HeadTag/HeadTag.js]: reverted

File: app/components/molecules/Testimonials/Testimonials.js (hunk 3 above)
Comment (chandrakailash1993) [reply app/components/molecules/Testimonials/Testimonials.js]: yes

File: app/components/templates/ErrorPage/ErrorPage.js (hunk 4 above)
Comment (chandrakailash1993) [reply app/components/templates/ErrorPage/ErrorPage.js]: done

File: app/components/templates/HomePage/HomePage.js (hunk 5 above)
Comment (chandrakailash1993) [reply app/components/templates/HomePage/HomePage.js]: done

File: app/lib/ContentSlotUnsafeHTML/ContentSlotUnsafeHTML.js (hunk 6 above)
Comment (chandrakailash1993) [reply app/lib/ContentSlotUnsafeHTML/ContentSlotUnsafeHTML.js]: done

File: app/components/molecules/NavBar/NavBar.js [hunk 7]
```diff
@@ -8,7 +9,7 @@ import type { Props } from './types';
 const NavBar = ({ className }: Props): Node => (
   <nav className={className}>
     <Link href="/error">
-      <a>Hello</a>
+      <a href="/error">Hello</a>
```
Comment (vinodloha) [inline app/components/molecules/NavBar/NavBar.js]: @chandrakailash1993 this is not needed, better put ignore statement.

File: app/components/atoms/Iframe/Iframe.js [hunk 8]
```diff
@@ -57,12 +67,14 @@ class Iframe extends PureComponent<Props, State> {
...
-        height={this.props.takeContentHeight ? this.state.frameHeight : height}
+        height={takeContentHeight ? frameHeight : height}
         frameBorder={frameBorder}
         onLoad={event => {
           if (takeContentHeight) {
+            // $FlowFixMe eslint-disable-line
+            const { document } = contentWindow;
             this.setState({
-              frameHeight: `${this.iframe.contentWindow.document.body.scrollHeight}px`,
+              frameHeight: `${document ? document.body.scrollHeight : 0}px`,
```
Comment (vinodloha) [inline app/components/atoms/Iframe/Iframe.js]: @chandrakailash1993 the code you changed and what was before are not the same things.

File: package.json [hunk 9]
```diff
@@ -86,6 +88,7 @@
     "lint-staged": "^7.3.0",
     "mini-css-extract-plugin": "^0.4.3",
     "node-sass": "^4.9.3",
+    "npm-run-all": "^4.1.5",
```
Comment (vinodloha) [inline package.json]: remove this package and its uses, it has a known venerability of using fs-events package

**Comment by vinodloha** [comment]
@chandrakailash1993 can you merge this with master, lot of changes there reached already.

File: package.json (hunk 9 above)
Comment (chandrakailash1993) [reply package.json]: done

File: app/components/atoms/Iframe/Iframe.js (hunk 8 above)
Comment (chandrakailash1993) [reply app/components/atoms/Iframe/Iframe.js]: i will be using this from component library in next PR, there I will change it

File: app/components/molecules/NavBar/NavBar.js (hunk 7 above)
Comment (chandrakailash1993) [reply app/components/molecules/NavBar/NavBar.js]: done

File: app/components/organisms/Header/Header.js [hunk 10]
```diff
@@ -2,16 +2,20 @@
...
+
 import type { Props } from './types';
 import NavBar from '../../molecules/NavBar';
 
 const Header = ({ className }: Props): Node => (
   <div className={className}>
     <header id="header">
       <Link href="/">
-        <a className="logo">Sample Logo</a>
+        <a href="/" className="logo">
```
Comment (vinodloha) [inline app/components/organisms/Header/Header.js]: Don't put href on <A> Link does that for you.

**Pull request closed without merging**